roles/openshift_health_checker/test/disk_availability_test.py

```diff
@@ -173,7 +173,62 @@ def test_fails_with_insufficient_disk_space(group_names, configured_min, ansible
 
     assert result['failed']
     for word in 'below recommended'.split() + extra_words:
-        assert word in result['msg']
+        assert word in result.get('msg', '')
+
+
+@pytest.mark.parametrize('group_names,context,ansible_mounts,failed,extra_words', [
+    (
+        ['nodes', 'masters'],
+        "upgrade",
+        [{
+            'mount': '/',
+            # not enough space for a master
+            # even under an "upgrade" context
+            'size_available': 1 * 10**9 + 1,
+            'size_total': 21 * 10**9 + 1,
+        }],
+        True,
+        ["1.0 GB"],
+    ),
+    (
+        ['nodes', 'masters'],
+        "upgrade",
+        [{
+            'mount': '/',
+            # not enough space for a master,
+            # but enough under "upgrade" context
+            'size_available': 10 * 10**9 + 1,
+            'size_total': 21 * 10**9 + 1,
+        }],
+        False,
+        [],
+    ),
+    (
+        ['nodes', 'masters'],
+        "health",
+        [{
+            'mount': '/',
+            # not enough space for a master,
+            # "health" context should not lower requirement
+            'size_available': 20 * 10**9 + 1,
+        }],
+        True,
+        ["20.0 GB", "below minimum"],
+    ),
+])
+def test_min_required_space_decreases_with_upgrade_context(group_names, context, ansible_mounts, failed, extra_words):
```

Comment:
Here and other places we say "decrease" or otherwise give suggestions as to what the values are compared to "non-upgrade". I'd avoid doing that type of comment or naming, because what governs the relationship of the numbers is actual data. Change the data and all the names become misleading.

Likewise, on an extreme, it is considered bad practice to name variables like `two = 2` only to see `two = 3` some day in the future...